Dockerfile

```diff
@@ -0,0 +1,13 @@
+# build stage
+FROM node:lts-alpine as build-stage
+WORKDIR /app
+COPY package*.json ./
+RUN npm install
+COPY . .
+RUN npm run build
+
+# production stage
+FROM nginx:stable-alpine as production-stage
+COPY --from=build-stage /app/dist /usr/share/nginx/html
+EXPOSE 80
+CMD ["nginx", "-g", "daemon off;"]
```

Comment:
should these be run based on whether we are running locally or remotely or as these to be run for each env? Sorry I didn't understand what does production stage mean here given the npm run build should ideally run vue right? Also, any reference for this?

Reply:
I just updated it. We're using multi-stage feature in docker to have configurations set for all types of envs from single place. Earlier, people used to have different dockerfile for every environment.

For VueJS, I've used the official guide for VueJS2
https://vuejs.org/v2/cookbook/dockerize-vuejs-app.html